packages/core/admin/admin/src/pages/MarketplacePage/tests/index.test.js

```diff
@@ -52,30 +40,23 @@ const setup = (props) => ({
       });
 
       return (
-        <Provider
-          store={createStore((state) => state, {
-            admin_app: { permissions: fixtures.permissions.app },
-          })}
-        >
-          <QueryClientProvider client={client}>
-            <TrackingProvider>
-              <IntlProvider locale="en" messages={{}} textComponent="span">
-                <ThemeProvider theme={lightTheme}>
-                  <Router history={history}>{children}</Router>
-                </ThemeProvider>
-              </IntlProvider>
-            </TrackingProvider>
-          </QueryClientProvider>
-        </Provider>
+        <QueryClientProvider client={client}>
+          <IntlProvider locale="en" messages={{}} textComponent="span">
+            <ThemeProvider theme={lightTheme}>
+              <NotificationsProvider>
+                <MemoryRouter>{children}</MemoryRouter>
+              </NotificationsProvider>
+            </ThemeProvider>
+          </IntlProvider>
+        </QueryClientProvider>
       );
     },
   }),
-
   user: userEvent.setup(),
 });
 
 const waitForReload = async () => {
-  await screen.findByTestId('marketplace-results');
+  await waitFor(() => expect(screen.queryByText('Loading content...')).not.toBeInTheDocument());
```

Comment:
What's the reason for waitFor + queryBy instead of findBy?

Reply:
finding by testId should be a last resort and realistically we want to be checking that the loader has finished before moving on because then we know the async calls are finished – which was the source of the error logging in these test suites :)

Reply:
Yes I definitely agree about test ids. I thought findByText could be used to [avoid the waitFor](https://testing-library.com/docs/dom-testing-library/api-async/#findby-queries) but actually it was made to make sure something appears, not that it disappears 🙃